### ISSUE TITLE
Expose IOx `tag`, `field` and `time` data type information (via SQL)

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#5659]
# TLDR
Expose the tag, field and time data type information from influx line protocol via a new SQL system table

# Background
IOx maps the influxdb data model to columns. For example this line protocol

```text
h2o_feet,location=coyote_creek water_level=3.048,level\ description="between 3 and 6 feet" 1566014040
h2o_feet,location=coyote_creek water_level=2.943,level\ description="below 3 feet" 1566014400
h2o_feet,location=coyote_creek water_level=2.831,level\ description="below 3 feet" 1566014760
```

Maps to a `h2o_feet` table with columns:
*  `location`
* `water_level`
* `description`
* `time`.  

```sql
org_bucket> select * from h2o_feet limit 5;
+----------------------+--------------+----------------------+-------------+
| level description    | location     | time                 | water_level |
+----------------------+--------------+----------------------+-------------+
| between 6 and 9 feet | coyote_creek | 2019-08-18T00:00:00Z | 8.504       |
| between 6 and 9 feet | coyote_creek | 2019-08-18T00:06:00Z | 8.419       |
| between 6 and 9 feet | coyote_creek | 2019-08-18T00:12:00Z | 8.32        |
| between 6 and 9 feet | coyote_creek | 2019-08-18T00:18:00Z | 8.225       |
| between 6 and 9 feet | coyote_creek | 2019-08-18T00:24:00Z | 8.13        |
+----------------------+--------------+----------------------+-------------+
```

In influxdb line protocol, `location` is are `tag` and `level description` is a `string field` and `water_level` is a `float field`

In flux and influxql `tags` and `fields` have different semantic meanings and the IOx catalog stores if a column is a tag or field (or timestamp). 

# workaround 

You can get this information indirectly today by looking at the data_type in the `information_schema` columns;

```sql
org_bucket> select column_name, data_type from information_schema.columns where table_name = 'h2o_feet';
+-------------------+-----------------------------+
| column_name       | data_type                   |
+-------------------+-----------------------------+
| level description | Utf8                        |
| location          | Dictionary(Int32, Utf8)     |
| time              | Timestamp(Nanosecond, None) |
| water_level       | Float64                     |
+-------------------+-----------------------------+
Returned 4 rows in 27.815688ms
```

Mapping the types:

```sql
org_bucket> 
select 
  column_name, 
  data_type, 
  case data_type 
    when 'Dictionary(Int32, Utf8)' then 'tag' 
    when 'Utf8' then 'field(string)'
    when 'Float64' then 'field(float)'
    when 'Int64' then 'field(int)'
    when 'UInt64' then 'field(uint)'
    when 'Boolean' then 'field(bool)'
    when 'Timestamp(Nanosecond, None)' then 'time'
    else 'UNKNOWN'
   end as 'influx_type' 
from 
  information_schema.columns 
where 
  table_name = 'h2o_feet';
+-------------------+-----------------------------+---------------+
| column_name       | data_type                   | influx_type   |
+-------------------+-----------------------------+---------------+
| level description | Utf8                        | field(string) |
| location          | Dictionary(Int32, Utf8)     | tag           |
| time              | Timestamp(Nanosecond, None) | time          |
| water_level       | Float64                     | field(float)  |
+-------------------+-----------------------------+---------------+
Returned 4 rows in 9.939278ms
```

# Proposed feature

It would be nice to provide an explicit system table that had the information directly, such as `system.columns`:

```sql
select * from system.columns
...
            "+---------------+------------+-------------+-------------+---------------+",
            "| partition_key | table_name | column_name | column_type | influxdb_type |",
            "+---------------+------------+-------------+-------------+---------------+",
            "| p1            | t1         | c1          | I64         | Tag           |",
            "| p1            | t1         | c2          | I64         | Field         |",
            "| p1            | t1         | c3          | String      |               |",
            "| p1            | t1         | time        | I64         | Timestamp     |",
            "+---------------+------------+-------------+-------------+---------------+",
        ];
```

# Context
IOx OG had this system table -- see https://github.com/influxdata/influxdb_iox/blob/4813cc8332256d47bac888ea1d2dfaffaf0f25fb/db/src/system_tables/columns.rs#L352-L361 -- which potentially could be used as a basis of implementation

# Implementation effort estimate
Andrew: 1 day
